chalice/app.py

```diff
@@ -765,3 +779,82 @@ class AuthRoute(object):
     def __init__(self, path, methods):
         self.path = path
         self.methods = methods
+
+
+class EventSource(object):
```

Comment:
It's intentional to have an `EventSource` type.  The idea would be this is used for all event types as support more types is added.  I don't envision adding a new public attr to `chalice.Chalice` for every new event source, that wouldn't be maintainable.  I can extract out a separate `CloudWatchEventSource` class with `EventSource` as its base type.  Seemed like overkill for now, but I don't mind adding it.